File: php/program/KitchenAppliances.php
<?php
require_once "Barang.php";

class KitchenAppliances extends Barang {
    protected $dayaListrik;
    protected $berat;
    protected $warna;
    protected $tipe;

    public function __construct($kode, $nama, $harga, $merk, $garansi, $stok, $gambar,
                                $dayaListrik, $berat, $warna, $tipe) {
        parent::__construct($kode, $nama, $harga, $merk, $garansi, $stok, $gambar);
        $this->dayaListrik = $dayaListrik;
        $this->berat = $berat;
        $this->warna = $warna;
        $this->tipe = $tipe;
    }

    // Getter dan Setter untuk daya listrik
    public function getDayaListrik(): string {
        return $this->dayaListrik;
    }
    public function setDayaListrik(string $dayaListrik): void{
        $this->dayaListrik = $dayaListrik;
    }

    // Getter dan Setter untuk berat
    public function getBerat(): string {
        return $this->berat;
    }
    public function setBerat(string $berat): void{
        $this->berat = $berat;
    }

    // Getter dan Setter untuk warna
    public function getWarna(): string {
        return $this->warna;
    }
    public function setWarna(string $warna): void{
        $this->warna = $warna;
    }

    // Getter dan Setter untuk tipe
    public function getTipe(): string {
        return $this->tipe;
    }
    public function setTipe(string $tipe): void{
        $this->tipe = $tipe;
    }
}

File: cpp/program/KitchenAppliances.cpp
#include <iostream>
#include "Barang.cpp"

using namespace std;

// class KitchenAppliances (Child) dari class Barang (Parent)
class KitchenAppliances : public Barang{
    // atribut private
    protected:
        string DayaListrik; // atribut DayaListrik
        string Berat; // atribut Berat
        string Warna; // atribut Warna
        string Tipe; // atribut tipe 

    public:
        KitchenAppliances(){
        
        }
        
        // saat membuat constructor pada class child,
        // maka harus memanggil constructor pada class parent
        KitchenAppliances(string kode, string nama, 
        string harga, string merk, string garansi, string stok, string DayaListrik, string Berat, 
        string Warna, string Tipe) : Barang(kode, nama, harga, merk, garansi, stok){
            this->DayaListrik = DayaListrik;
            this->Berat = Berat;
            this->Warna = Warna;
            this->Tipe = Tipe;
        }

        string getDayaListrik(){
            return DayaListrik;
        }

        void setDayaListrik(string DayaListrik){
            this->DayaListrik = DayaListrik;
        }

        string getBerat(){
            return Berat;
        }

        void setBerat(string Berat){
            this->Berat = Berat;
        }

        string getWarna(){
            return Warna;
        }

        void setWarna(string Warna){
            this->Warna = Warna;
        }

        string getTipe(){
            return Tipe;
        }

        void setTipe(string Tipe){
            this->Tipe = Tipe;
        }
};